### PR TITLE
Add MacOS support

### PR DESCRIPTION
**Pull request by zakkak**
Implements part of https://github.com/graalvm/mandrel/issues/114

### CODE REVIEW

File: build.java [hunk 1]
```diff
@@ -171,20 +174,20 @@ public static void main(String... args) throws IOException
                 {
                     libchelperSource = Path.of("substratevm", "mxbuild", PLATFORM, "src", "com.oracle.svm.native.libchelper", ARCH, "liblibchelper.a");
                     libjvmSource = Path.of("substratevm", "mxbuild", PLATFORM, "src", "com.oracle.svm.native.jvm.posix", ARCH, "libjvm.a");
-                    reporterchelperSource = Path.of("substratevm", "mxbuild", PLATFORM, "src", "com.oracle.svm.native.reporterchelper", ARCH, "libreporterchelper.so");
+                    reporterchelperSource = Path.of("substratevm", "mxbuild", PLATFORM, "src", "com.oracle.svm.native.reporterchelper", ARCH, IS_MAC ? "libreporterchelper.dylib" : "libreporterchelper.so");
```
Comment (jerboaa) [inline build.java:177]: How about `System.mapLibraryName("reporterchelper")`?

File: build.java [hunk 2]
```diff
@@ -42,6 +42,7 @@
 {
     static final Logger logger = LogManager.getLogger(build.class);
     public static final boolean IS_WINDOWS = System.getProperty("os.name").matches(".*[Ww]indows.*");
+    public static final boolean IS_MAC = System.getProperty("os.name").matches(".*[Mm]ac.*");
```
Comment (jerboaa) [inline build.java:45]: No longer used anywhere?

**Comment by zakkak** [comment]
It looks like this needs some more work to package mandrel in the way that MacOS expects JDKs to be packed, i.e. move files under `Contents/Home` and include some additional mac-specific files and folders.

**Review by jerboaa — APPROVED** [review]
Nice work!

**Pull request merged**